extern/agg24-svn/include/agg_pixfmt_rgba.h

```diff
@@ -1515,13 +1515,12 @@ namespace agg
         typedef typename color_type::calc_type calc_type;
         enum 
         {
-            num_components = 4,
             pix_step = 4,
             pix_width = sizeof(value_type) * pix_step,
         };
         struct pixel_type
         {
-            value_type c[num_components];
+            value_type c[pix_step];
```

Comment:
I don't think this should change, as it makes the `rgba` file inconsistent with the `rgb` one, and semantically these are different things.

Reply:
What if we change them as well? Looks like this was code is simpler and compliant.
Then num_components is not used anywhere, but I can return if you like.
PTAL

Reply:
@QiCuiHub: Has this point been resolved? (I don't have any opinion either way, but I wanted to make sure this isn't a blocker.)

Reply:
(For what it's worth, if it is easier to land this overall fix without chaning the bounds here, I'd welcome that as a simpler first step; the deduplication of these two redundant enumerators could be a separate PR.)

Reply:
I like as small as possible patches as well. Still here to use "this + 1" we need array size to be the "pix_step" not "num_components". Then num_components is unused. So the best I can do is moving lines with "-            num_components = ..." into a separate PR.

Reply:
Not sure who you're tagging there; as you can see I've already approved.

Reply:
@vitalybuka: Ah of course, `Step` could be not 1... say, is the _current_ code correct when `Step` is not 1? In that case the current code's `next` would skip over some array elements, right?

Reply:
@QiCuiHub: Sorry, I thought this point was still unresolved; please ignore!